moflask/jwt.py

```diff
@@ -70,26 +70,28 @@ def organizations_with_access(self, admitted_roles: Iterable[str]) -> Set[str]:
             A set of organization strings for which the session has any of the given roles.
             If both a parent and a parent>sub organization would be part of the set, then only the
             parent is returned.
+            This is the minimal set of organizations so that has_any_role_of() would return True
+            for all of them and any of their sub-organizations.
         """
         roles = frozenset(admitted_roles)
         orgs = frozenset(o for o, r in self.roles.items() if not roles or r & roles)
         cleaned = frozenset(o for o in orgs if not any(p in orgs for p in iterate_parents(o)))
         return cleaned
 
     def organizations_with_inherited_access(self, admitted_roles: Iterable[str]) -> Set[str]:
-        """Get all organizations within which the session has any of the given roles.
+        """Get the ancestors of all organizations for which the session has any of the given roles.
 
-        This method is usually used to determine which ressources the current user inherits from
-        parent organizations. Usually access to such ressources is read-only.
+        This method is used to determine which ressources the current user inherits from parent
+        organizations. Usually access to such ressources is read-only.
 
         Returns:
-            A set of organization strings of all the organizations for which the session has any of
-            the given roles and all of their ancestors.
+            A set of organization strings of all the ancestors of organizations for which the
+            session has any of the given roles.
+            For any of the returned organizations has_any_role_of() would return False, but there
+            exists at least one descendant of the organization for which it would return True.
```

Comment:
Thanks, I think in this version + new comments it's a lot clearer what the methods are for :)

However, I still struggle a bit with the method names. "organizations with access" sounds like the returned organizations have access to something although it's the session that has access to the organizations for a specific role. The inheritance makes that even more confusing. "accessible_organizations" might be less ambiguous?

For the second method, I feel like what it actually does is combining iterate_parents() for more than 1 org. So maybe it could make code more readable to instead add a "bulk_iterate_parents" method that could be called on the result of "accessible_organizations" if needed?

Reply:
Just realized that in other apps the methods will usually be called as `session.organizations…` so that sounds a bit different :thinking: `session.organizations_for_roles()` and `session.parent_organizations_for_roles()` might even work better than `session.accessible_organizations` and `session.accessbile_parent_organizations`?

Reply:
Yes! I find it much easier now to grasp what the methods are doing :) 

Just having a hard time with one of the new sentences in the comment: "This means an empty list of roles behaves different from when an organization is not even mentioned in the session." I think I might have figured out what it means but it took me quite a while. IMHO it could be removed all together, I would think the previous sentence makes it clear enough to expect a non-empty return value when passing an empty list of roles?

Reply:
Ok, I’ve removed the sentence again.